src/activations.jl

```diff
@@ -235,22 +236,72 @@ for f in ACTIVATIONS
       error("Use broadcasting (`", $(string(f)), ".(x)`) to apply activation functions to arrays.")
 end
 
-## Define rrules for some activation functions, along with the 
+## Faster, less accurate, versions of some.
+
+"""
+    tanh_fast(x)
+
+This is a faster but slighly less accurate version of `tanh`.
+
+Where Julia's `tanh` function has an error under 2 eps, this
+may be wrong by 5 eps, a reduction by less than one decimal digit. 
+
+For `x::Float32` this is usually about 10 times faster,
+with a smaller speedup for `x::Float64`.
+"""
+@inline function tanh_fast(x::Float32)
+    x2 = abs2(x)
+    n = evalpoly(x2, (1.0f0, 0.1346604f0, 0.0035974074f0, 2.2332108f-5, 1.587199f-8))
+    d = evalpoly(x2, (1.0f0, 0.4679937f0, 0.026262015f0, 0.0003453992f0, 8.7767893f-7))
+    ifelse(x2 < 66f0, x * (n / d), sign(x))
+end
+
+@inline function tanh_fast(x::Real)
+    exp2x = @fastmath exp(x + x)
+    y = (exp2x - 1) / (exp2x + 1) 
+    # That has large errors near zero; using `expm1` would more accurate, but about as slow as `tanh`.
+    # Instead, we switch to an Taylor series. This is very accurate within its range:
+    x2 = x * x
+    ypoly = x * evalpoly(x2, (1.0, -0.33333333333333337, 0.1333333333332623, -0.0539682539194502, 0.021869476267930975, -0.00886184042142138, 0.0035188503873932893))
+    ifelse(x2 > 900.0, sign(y), ifelse(x2 < 0.02, oftype(y, ypoly), y))
+end
+
+tanh_fast(x::Float16) = Base.tanh(x) # Other approximations are very badly behaved for Float16; none are fast.
+
+"""
+    sigmoid_fast(x)
+
+This is a faster, and very slightly less accurate, version of `sigmoid`.
+"""
+@inline function sigmoid_fast(x::Real)
+    t = @fastmath exp(-abs(x))
+    y = ifelse(x ≥ 0, inv(1 + t), t / (1 + t))
+    ifelse(x > 40, one(y), ifelse(x < -80, zero(y), y))
+end
```

Comment:
For Float32, this `sigmoid_fast` isn't as big an improvement as `tanh_fast`. That rational function symmetric about zero gives you terrible relative accuracy for negative `x`, once it's going to zero. 
```
julia> xs = collect(0.01:0.001:10); ys = similar(xs);

julia> xs_32 = Float32.(collect(xs)); ys_32 = similar(xs_32);

# Float32

julia> @btime @. $ys_32 = sigmoid($xs_32);
  26.250 μs (0 allocations: 0 bytes)

julia> @btime @. $ys_32 = sigmoid_fast($xs_32);
  7.135 μs (0 allocations: 0 bytes)

julia> @btime @. $ys_32 = sigmoid_fast_old($xs_32);  # version with polynomial from tanh_fast
  3.870 μs (0 allocations: 0 bytes)

# Float64

julia> @btime @. $ys = sigmoid($xs);
  34.250 μs (0 allocations: 0 bytes)

julia> @btime @. $ys = sigmoid_fast($xs);
  12.000 μs (0 allocations: 0 bytes)
```
Compare tanh:
```
julia> @btime @. $ys_32 = tanh($xs_32);
  27.125 μs (0 allocations: 0 bytes)

julia> @btime @. $ys_32 = tanh_fast($xs_32);
  3.594 μs (0 allocations: 0 bytes)

julia> @btime @. $ys = tanh($xs);
  44.958 μs (0 allocations: 0 bytes)

julia> @btime @. $ys = tanh_fast($xs);
  17.041 μs (0 allocations: 0 bytes)
```